util/chunk/checksum.go

```diff
@@ -0,0 +1,120 @@
+package chunk
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"io"
+	"os"
+)
+
+const (
+	checksumBlockSize   = 1024
+	checksumSize        = 4
+	checksumPayloadSize = checksumBlockSize - checksumSize
+)
+
+type checksum struct {
+	disk        *os.File
+	buf         []byte
+	payload     []byte
+	payloadUsed int
+	size        int64
```

Comment:
Recording size in the struct is not good, so that I can't open an exists checksum file and use it.